web/shared/src/typeahead.ts

```diff
@@ -46,8 +46,15 @@ type UnicodeEmoji = {
     is_realm_emoji: false;
 };
 
+const remove_diacritics_cache = new Map<string, string>();
+
 export function remove_diacritics(s: string): string {
-    return s.normalize("NFKD").replace(unicode_marks, "");
+    if (remove_diacritics_cache.has(s)) {
+        return remove_diacritics_cache.get(s)!;
+    }
```

Comment:
This would be better written as doing a `.get` and returning it if the value is defined; that would avoid doing two lookups in the case of a cache hit unnecessarily.

Reply:
oh of course

Reply:
updated :)